ethers-solc/src/remappings.rs

```diff
@@ -0,0 +1,284 @@
+///
+/// contract MyContract {
+///     using SafeMath for uint256;
+///     ...
+/// }
+/// ```
+///
+/// When using solc, you have to specify the following:
+///
+/// "prefix" = the path that's used in your smart contract, i.e.
+/// "@openzeppelin/contracts-ethereum-package" "target" = the absolute path of OpenZeppelin's
+/// contracts downloaded on your computer
+///
+/// The format looks like this:
+/// `solc prefix=target ./MyContract.sol`
+///
+/// solc --bin
+/// @openzeppelin/contracts-ethereum-package=/Your/Absolute/Path/To/@openzeppelin/
+/// contracts-ethereum-package ./MyContract.sol
+///
+/// [Source](https://ethereum.stackexchange.com/questions/74448/what-are-remappings-and-how-do-they-work-in-solidity)
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord)]
+pub struct Remapping {
+    pub name: String,
+    pub path: String,
+}
+
+impl Serialize for Remapping {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for Remapping {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        let remapping = String::deserialize(deserializer)?;
+        let mut split = remapping.split('=');
+        let name = split
+            .next()
+            .ok_or_else(|| serde::de::Error::custom("no remapping prefix found"))?
+            .to_string();
+        let path = split
+            .next()
+            .ok_or_else(|| serde::de::Error::custom("no remapping path found"))?
+            .to_string();
+        Ok(Remapping { name, path })
+    }
+}
+
+// Remappings are printed as `prefix=target`
+impl fmt::Display for Remapping {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}={}", self.name, self.path)
+    }
+}
+
+impl Remapping {
+    /// Detects a remapping prioritizing Dapptools-style remappings over `contracts/`-style ones.
+    fn find(root: &str) -> Result<Self> {
+        Self::find_with_type(root, DAPPTOOLS_CONTRACTS_DIR)
+            .or_else(|_| Self::find_with_type(root, JS_CONTRACTS_DIR))
+    }
+
+    /// Given a path and the style of contracts dir, it proceeds to find
+    /// a `Remapping` for it.
+    fn find_with_type(name: &str, source: &str) -> Result<Self> {
+        let pattern = if name.contains(source) {
+            format!("{}/**/*.sol", name)
+        } else {
+            format!("{}/{}/**/*.sol", name, source)
+        };
+        let mut dapptools_contracts = glob::glob(&pattern)?;
+        let next = dapptools_contracts.next();
+        if next.is_some() {
+            let path = format!("{}/{}/", name, source);
+            let mut name = name.split('/').last().unwrap().to_string();
+            name.push('/');
+            Ok(Remapping { name, path })
```

Comment:
since I already pulled in `walkdir`, ref `util::source_files`, I wonder if this can be replaced that. 
We could extract the walkdir iter part to a new function `util::source_files_iter` maybe

Reply:
As discussed, I tried this and it was not obvious how to do. Let's refactor in a follow-up PR if needed.